src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/unit/uom/service/impl/UOMServiceImpl.java

```diff
@@ -375,6 +387,31 @@ public void updateExistingUOMByCode(String code, UOMDtoV2 sourceUOMDto) throws U
         log.info("Updated UOMEntity with id: {}", uomEntity.getId());
     }
 
+
+    @Transactional
+    @Override
+    public void updateExistingUOMByCode(String code, JsonPatch jsonPatch) throws UOMException, JsonPatchException, JsonProcessingException {
+        Optional<UOMEntity> optionalUOMEntity = uomJpaRepository.findByCode(code);
+        if(optionalUOMEntity.isEmpty()) {
+            throw new UOMException(WMSErrorCode.WMS_NOT_FOUND, new Object[]{code});
+        }
+        log.debug("UOM does exists with code: {}", code);
+        UOMEntity uomEntity = optionalUOMEntity.get();
+        UOMDtoV2 targetUOMDto = uomEntityToDtoV2Converter.convert(uomEntity);
+        JsonNode patchedUOMNode = null;
+        UOMDtoV2 patchedUOMDto = null;
+        try {
+            patchedUOMNode = jsonPatch.apply(objectMapper.convertValue(targetUOMDto, JsonNode.class));
+            patchedUOMDto = objectMapper.treeToValue(patchedUOMNode, UOMDtoV2.class);
+        } catch (InvalidFormatException e) {
+            throw new UOMException(WMSErrorCode.WMS_ATTRIBUTE_INVALID, new Object[]{"Invalid value Provided"});
```

Comment:
- log the exception at error level
- provide the field name or value instead of the string "Invalid value Provided" in the exception parameter
- look into how WMSErrorCode.WMS_ATTRIBUTE_INVALID is parsed in the exception handler